java/vector/src/main/java/org/apache/arrow/vector/util/ValueVectorUtility.java

```diff
@@ -82,4 +89,43 @@ public static String getToString(ValueVector vector, int start, int end) {
 
     return sb.toString();
   }
+
+  /**
+   * Validate field vector.
+   */
+  public static void validate(FieldVector vector) {
+    Preconditions.checkNotNull(vector);
+
+    ArrowType arrowType = vector.getField().getType();
+    int typeBufferCount = TypeLayout.getTypeBufferCount(arrowType);
+    TypeLayout typeLayout = TypeLayout.getTypeLayout(arrowType);
+
+    if (vector.getValueCount() < 0) {
+      throw new RuntimeException("vector valueCount is negative");
+    }
+
+    if (vector.getFieldBuffers().size() != typeBufferCount) {
+      throw new RuntimeException(String.format("Expected %s buffers in vector of type %s, got %s",
+          typeBufferCount, vector.getField().getType().toString(), vector.getBufferSize()));
+    }
+
+    for (int i = 0; i < typeBufferCount; i++) {
+      ArrowBuf buffer = vector.getFieldBuffers().get(i);
+      BufferLayout bufferLayout = typeLayout.getBufferLayouts().get(i);
+      if (buffer == null) {
+        continue;
+      }
+      int minBufferSize = vector.getValueCount() * bufferLayout.getTypeBitWidth();
+
+      if (buffer.capacity() < minBufferSize / 8) {
```

Comment:
for offset buffers, we need to have:

int minBufferSize = (vector.getValueCount() + 1) * bufferLayout.getTypeBitWidth();

Reply:
Thanks, right, I added offset buffer check for variableWidthVector and ListVector in ValueVectorVisitor (rather than add the check here)